test/link_main_test.cc

```diff
@@ -6,3 +6,5 @@ void BM_empty(benchmark::State& state) {
   }
 }
 BENCHMARK(BM_empty);
+
+int main() {}
```

Comment:
this is surprising. the whole point of the test is to check that linking main works without this (https://github.com/google/benchmark/blob/main/test/CMakeLists.txt#L128).  if that's broken then something else has gone wrong.

Reply:
I see. `main` is optimized away by `-flto` and also needs to be exported. Fixed.